Mathlib/Algebra/Group/Defs.lean

```diff
@@ -391,6 +401,8 @@ class RightCancelSemigroup (G : Type u) extends Semigroup G where
 #align right_cancel_semigroup.ext_iff RightCancelSemigroup.ext_iff
 #align right_cancel_semigroup.ext RightCancelSemigroup.ext
 
+attribute [instance 75] RightCancelSemigroup.toSemigroup -- See note [lower cancel priority]
```

Comment:
What's the default priority here? Is it 1000 or 100?

Reply:
All instances, also the "inheritance" ones generated by `extends`, by default get a priority of 1000. Some of the manually defined inheritance instances have already been assigned priority 100, so IMO we should follow mathlib3 principles and stick with 100 for inheritance plus or minus a bonus. I chose 75 because that is also used in `SetLike`'s hierarchy.

Reply:
Can we change core to set everything to 100 automatically? I worry that we're just making a mess tweaking everything manually if the defaults are still far from desirable.

Reply:
One thing that I'm wondering about is: should we even bother with instance priority = 100, given that instances are now matched by discrimination tree first? I don't know enough about Lean 4 synthesis at scale to say for sure. I do know, however, since there are instances with priority 100 already, these instances should have a lower priority than those, and this change is making things work that should work.

Reply:
> given that instances are now matched by discrimination tree first?

In that an instance of the form `SomeClass (Foo _)` if it matches is always tried first over `SomeClass (_)`, even if it has lower priority? Or is the tree just used to break ties?

Reply:
Looks like the tree is not used to break ties at all!
```
class Foo (α : Type)
class Bar (α : Type) extends Foo α -- Try moving this line to after the `attribute` line

instance : Foo ℤ := ⟨⟩

attribute [instance 1] instFoo -- Try commenting out this line

set_option trace.Meta.synthInstance true
example : Foo ℤ := by infer_instance
```